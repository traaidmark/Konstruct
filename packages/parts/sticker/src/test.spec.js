// PARTS [ DATA > TESTS ] ######################################################

// 1. DEPENDENCIES +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 1.1. EXTERNAL DEPENDENCIES ..................................................

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

// 1.1. END ....................................................................

// 1.2. INTERNAL DEPENDENCIES ..................................................

import Sticker from '.';

// 1.2. END ....................................................................

// 1. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 2. PROPS ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

const props = {
  prefix: 'usd',
  value: '2300.00',
  label: 'Amount',
  placement: 'top',
  align: 'left',
  size: 'large'
}

// 2. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// 3. TEST +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

describe('Parts: Sticker', () => {

  // 3.1. RENDER WITHOUT PROPS .................................................

  test('renders correctly without any props', () => {
    const tree = renderer.create(
      <Sticker />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

  // 3.2. RENDER WITH ALL PROPS ................................................

  test('renders correctly with all props', () => {
    const tree = renderer.create(
      <Sticker
        prefix={ props.prefix }
        value={ props.value }
        size={ props.size }
        align={ props.align }
        placement={ props.placement }
        label={ props.label }
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // 3.1. END ..................................................................

});

// 3. END ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

// END OF FILE #################################################################
