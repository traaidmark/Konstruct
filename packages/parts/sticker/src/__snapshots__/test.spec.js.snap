// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parts: Sticker renders correctly with all props 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: var(--weight-bold);
  color: var(--color-darker);
}

.c1 strong {
  font-size: var(--font-large-size);
  line-height: var(--font-large-lineHeight);
}

.c1 i {
  font-weight: var(--weight-light);
  color: var(--color-dark);
  font-style: normal;
  text-transform: uppercase;
  margin-right: 5px;
  font-size: var(--font-large-size);
  line-height: var(--font-large-lineHeight);
}

.c0 {
  --sticker-label-color: var(--color);
  display: block;
  text-align: left;
}

.c0 small {
  display: block;
  font-weight: var(--weight-regular);
  color: var(--sticker-label-color);
}

<div
  className="c0"
>
  <small>
    Amount
  </small>
  <div
    className="c1"
    size="large"
  >
    <i>
      usd
    </i>
    <strong>
      2300.00
    </strong>
  </div>
</div>
`;

exports[`Parts: Sticker renders correctly without any props 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: var(--weight-bold);
  color: var(--color-darker);
}

.c1 strong {
  font-size: var(--font-medium-size);
  line-height: var(--font-medium-lineHeight);
}

.c1 i {
  font-weight: var(--weight-light);
  color: var(--color-dark);
  font-style: normal;
  text-transform: uppercase;
  margin-right: 5px;
  font-size: var(--font-normal-size);
  line-height: var(--font-normal-lineHeight);
}

.c0 {
  --sticker-label-color: var(--color);
  display: block;
  text-align: center;
}

.c0 small {
  display: block;
  font-weight: var(--weight-regular);
  color: var(--sticker-label-color);
}

<div
  className="c0"
>
  <div
    className="c1"
    size="default"
  >
    
    <strong>
      
    </strong>
  </div>
  <small>
    
  </small>
</div>
`;
